channels/cliprdr/cliprdr_common.c

```diff
@@ -241,141 +241,85 @@ wStream* cliprdr_packet_file_contents_response_new(const CLIPRDR_FILE_CONTENTS_R
 	return s;
 }
 
+static const char* bool_str_true = "true";
+static const char* bool_str_false = "false";
+static const char* boolstr(BOOL rc)
+{
+	return rc ? bool_str_true : bool_str_false;
+}
+
 wStream* cliprdr_packet_format_list_new(const CLIPRDR_FORMAT_LIST* formatList,
-                                        BOOL useLongFormatNames)
+                                        BOOL useLongFormatNames, BOOL useAsciiNames)
 {
-	wStream* s = NULL;
-	size_t formatNameSize = 0;
-	char* szFormatName = NULL;
-	WCHAR* wszFormatName = NULL;
-	BOOL asciiNames = FALSE;
-	CLIPRDR_FORMAT* format = NULL;
-	UINT32 length = 0;
+	WINPR_ASSERT(formatList);
 
 	if (formatList->common.msgType != CB_FORMAT_LIST)
 		WLog_WARN(TAG, "called with invalid type %08" PRIx32, formatList->common.msgType);
 
-	if (!useLongFormatNames)
-	{
-		length = formatList->numFormats * 36;
-		s = cliprdr_packet_new(CB_FORMAT_LIST, 0, length);
-
-		if (!s)
-		{
-			WLog_ERR(TAG, "cliprdr_packet_new failed!");
-			return NULL;
-		}
-
-		for (UINT32 index = 0; index < formatList->numFormats; index++)
-		{
-			size_t formatNameLength = 0;
-			format = (CLIPRDR_FORMAT*)&(formatList->formats[index]);
-			Stream_Write_UINT32(s, format->formatId); /* formatId (4 bytes) */
-			formatNameSize = 0;
-
-			szFormatName = format->formatName;
-
-			if (asciiNames)
-			{
-				if (szFormatName)
-					formatNameLength = strnlen(szFormatName, 32);
+	if (useLongFormatNames && useAsciiNames)
+		WLog_WARN(TAG,
+		          "called with invalid arguments useLongFormatNames=%s, useAsciiNames requires "
+		          "useLongFormatNames=false, ignoring argument.",
+		          boolstr(useLongFormatNames));
```

Comment:
Would be easier to write `useLongFormatNames=true` instead of introducing `boolstr` and using format parameters.